src/Module.ts

```diff
@@ -175,6 +186,8 @@ export default class Module {
 			this.dynamicImports = [];
 			this.dynamicImportResolutions = [];
 		}
+		this.isEntryPoint = false;
+		this.execIndex = undefined;
```

Comment:
It's usually a good practice in JavaScript to set all object properties upfront to avoid reshaping - in this scenario we are creating a a lot of Module objects, before then later running a batch operation on all which will set this property, which would otherwise result in a bulk instance shape change at that point. Reshaping of a class will usually converge over time so you can often get away without it, but because of the upfront initialization of all Module objects first, the reshaping here can benefit from it (the best example I could find https://jsperf.com/initializing-properties-to-undefined, although it's probably not necessarily indicative)

Reply:
Agreed. Though I would actually assign `null` instead of `undefined` to make this more explicit, the same way it is done e.g. in the ESTree spec. That also reduces the risk of this being accidentally removed as a seeming no-op (which of course it isn't)

Reply:
I usually try to avoid null for the `typeof null === 'object'` catch but sure.